app/models/spree/page.rb

```diff
@@ -51,4 +54,8 @@ def update_positions_and_slug
   def not_using_foreign_link?
     foreign_link.blank?
   end
+
+  def generate_slug
+    self.slug ||= title.downcase.gsub(/\W+/, '-').gsub(%r{(^/?\-+|\-+$)}, '')
```

Comment:
would `parameterize` be a good fit here?

Reply:
That's perfect, thanks!